### PR TITLE
Add recommended content to dialogue match form and restyle

### PR DESCRIPTION
**Pull request by lagerros**
* Add a card to the dialogues match form containing recommended content by the other user (top tags, posts, comments)
* Restyle the whole dialogue match form to look cleaner given this
* Fix some dependency cycles that resulted

Before: 
image

After: 
image

### CODE REVIEW

**Comment by rjmk** [comment]
@lagerros can we do an AB Test here?

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx [hunk 1]
```diff
@@ -814,10 +971,13 @@ const NextStepsDialog = ({ onClose, userId, targetUserId, targetUserDisplayName,
   return (
     <LWDialog open onClose={onClose} className={classes.mobileDialog}>
       <div className={classes.dialogBox}>
-        <DialogTitle className={classes.dialogueTitle}>Alright, you matched with {targetUserDisplayName}!</DialogTitle>
+        <DialogTitle className={classes.dialogueTitle}><span className={classes.matchHeader}>(Match)</span> {targetUserDisplayName}</DialogTitle>
         <DialogContent >
+          <UserRecommendedContent targetUserId={targetUserId} classes={classes} />
+          <h3 className={classes.sectionHeader}>Topics</h3>
+          {/* <div>Check any suggestion you're interested in, or add your own.</div> */}
```
Comment (b0b3rt) [inline packages/lesswrong/components/users/DialogueMatchingPage.tsx:978]: does this need to be cleaned up?

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx [hunk 2]
```diff
@@ -679,6 +734,107 @@ const Headers = ({ titles, classes, headerClasses }: { titles: string[], headerC
...
+      <div className={classes.cardTitle}>
+        Recommended content
+      </div>
+      <div className={classNames(classes.recommendedContentContainer, {
+        [classes.recommendedContentContainerExpandedMobile]: isExpanded})}>
+        <div className={classes.contentRecommendationsList}>
+          {allRecommendations.slice(0, numShown).map( (item, index) => <TopicSuggestion key={index} reactIconName={item.reactIconName} prefix={item.prefix} Content={item.Content} isExpanded={isExpanded} />) } 
+        </div>
+        <div className={classes.recommendedContentRightContainer}>
+          {(allRecommendations && allRecommendations.length > 0) && <ExpandCollapseText isExpanded={isExpanded} numHidden={numHidden} toggleExpansion={toggleExpansion} />}
```
Comment (b0b3rt) [inline packages/lesswrong/components/users/DialogueMatchingPage.tsx:831]: `(allRecommendations && allRecommendations.length > 0)` feels unnecessary given the check at L817

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx [hunk 3]
```diff
@@ -679,6 +734,107 @@ const Headers = ({ titles, classes, headerClasses }: { titles: string[], headerC
...
+`, {
+  variables: { userId: currentUser?._id, targetUserId: targetUserId, limit : 3 },
+  skip: !currentUser
+});
+
+const topTags:TagWithCommentCount[] | undefined = tagData?.UserTopTags;
+const readPosts:PostYouveRead[] | undefined = postsData?.UsersReadPostsOfTargetUser
+const recommendedComments:RecommendedComment[] | undefined = commentsData?.UsersRecommendedCommentsOfTargetUser
+
+if (!currentUser || !topTags || !readPosts || !recommendedComments) return <Loading />;
```
Comment (b0b3rt) [inline packages/lesswrong/components/users/DialogueMatchingPage.tsx:797]: Is this (and most of the preceeding logic) copied from somewhere?  Looks very familiar

**Review by rjmk — DISMISSED** [review]
We've got a bunch of different ideas for changing this form incoming. I want to AB Test the different changes, including the null change. Could you start the sequence of AB Tests by creating one for this PR?

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 4]
```diff
@@ -419,13 +432,15 @@ const DialogueRecommendationRow = ({ targetUser, checkId, userIsChecked, userIsM
 
 const DialogueRecommendationRowComponent = registerComponent('DialogueRecommendationRow', DialogueRecommendationRow, {styles});
 const TopicSuggestionComponent = registerComponent('TopicSuggestion', TopicSuggestion, {styles});
+const CommentViewComponent = registerComponent('CommentView', CommentView, {styles});
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:435]: If you're now registering the `CommentView` component and wrapping it in the `styles` HoC, but it previously took `classes` as a prop, that means wherever you were using it before, you were passing in `classes` to it manually.  I don't see any diffs that reflect that having been changed/fixed so I'm a bit suspicious that something is broken here

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx [hunk 5]
```diff
@@ -415,16 +415,23 @@ const styles = (theme: ThemeType) => ({
...
+    color: theme.palette.lwTertiary.main,
     paddingBottom: 8
   },
   dialogueFormatGrid: {
     display: 'grid',
     grid: 'auto-flow / 1fr 40px 40px 40px',
     alignItems: 'center',
-    marginBottom: 8
+    marginBottom: 16,
+    rowGap: '6px',
```
[the remark this answers is not in the document]
Comment (lagerros) [reply packages/lesswrong/components/users/DialogueMatchingPage.tsx:426]: tried numbers, didn't work for a strange reason, had to use pixel values. can show you if you want

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx (hunk 3 above)
Comment (lagerros) [reply packages/lesswrong/components/users/DialogueMatchingPage.tsx:797]: it is. Considered putting it into a reusable hook, but, third times the charm

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx (hunk 2 above)
Comment (lagerros) [reply packages/lesswrong/components/users/DialogueMatchingPage.tsx]: legit

**Review by rjmk — COMMENTED** [review]
Hopefully this removes my request for feedback without creating an approval

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx (hunk 4 above)
Comment (lagerros) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:435]: It's only used in two places (frontpage recommendations, and dialogue match form). I checked behavior in both before making the PR, and it seemed fine. 

Nonetheless, I fixed the issue (properly imported CommentView and removed classes argument), and thanks for catching!

**Comment by rjmk** [comment]
Yeah I'm just keeping an eye and waiting for Robert's approving review

**Review by jpaddison3 — COMMENTED** [review]
No objection

**Review by b0b3rt — APPROVED** [review]

**Pull request merged**